csrc/dispatch.cpp

```diff
@@ -72,7 +69,8 @@ void Val::dispatch(T handler, Val* val) {
       ptr(handler)->handle(val->as<PipelineVal>());
       return;
     default:
-      break;
+      ptr(handler)->handleGeneric(val);
```

Comment:
Seems that this does not work. There do exist valid cases that override `handle(Val*)`, for example https://github.com/NVIDIA/Fuser/blob/965032565652ceb8c6ad5ac56fc8f87ed995386e/csrc/iter_visitor.cpp#L57-L62

Reply:
It seems to me that this is because `handle` is both used for control dispatch and processing of statements. For example, if we renamed `handle(Val*`)` to `dispatch(Val*)`, we could just have `handle(Val*)` instead of `handleGeneric(Val*)`. I think that would be clearer, but I'm ok with this design too.

Reply:
I think what you said makes sense. But I would prefer to do it in a separate PR, because if we rename `handle(Val*)`, we should also rename `handle(Expr*)` and `handle(Statement*)` for consistency. This is a big enough change and deserve a separate PR.